### PR TITLE
EMB-12126 - Get latest Embrace SDK Version

### PR DESCRIPTION
**Pull request by ghost**
Goal
Integrate a network API to get the latest Embrace SDK version
+add form manager and network layer

Testing
manual testing

Release Notes
WHAT:

WHY:

WHO:

### CODE REVIEW

File: src/main/kotlin/io/embrace/android/intellij/plugin/manager/EmbraceIntegrationDataProvider.kt [hunk 1]
```diff
@@ -0,0 +1,25 @@
+package io.embrace.android.intellij.plugin.manager
```
Comment (lucaslabari) [inline src/main/kotlin/io/embrace/android/intellij/plugin/manager/EmbraceIntegrationDataProvider.kt:1]: can we change this package name?

File: src/main/kotlin/io/embrace/android/intellij/plugin/network/HttpClient.kt [hunk 2]
```diff
@@ -0,0 +1,39 @@
+package io.embrace.android.intellij.plugin.network
+
+import java.io.IOException
+import java.net.URI
+import java.net.URISyntaxException
+import java.net.http.HttpClient
+import java.net.http.HttpRequest
+import java.net.http.HttpResponse
+
+class HttpClient : ApiClient {
```
Comment (lucaslabari) [inline src/main/kotlin/io/embrace/android/intellij/plugin/network/HttpClient.kt:10]: I think we should use another name so it doesn't create confusion with java.net.http.HttpClient

File: src/main/kotlin/io/embrace/android/intellij/plugin/ui/components/EmbBlockCode.kt [hunk 3]
```diff
@@ -1,11 +1,16 @@
...
 
+import io.embrace.android.intellij.plugin.manager.EmbraceIntegrationDataProvider
+import io.embrace.android.intellij.plugin.network.HttpClient
 import java.awt.Color
 import java.awt.Font
 import javax.swing.JLabel
 
 internal class EmbBlockCode(block: CODE_BLOCK) : JLabel() {
 
+    private val httpClient = HttpClient()
```
Comment (lucaslabari) [inline src/main/kotlin/io/embrace/android/intellij/plugin/ui/components/EmbBlockCode.kt:11]: I think we can inject this, so we can use the same instance in all components, forms. Or maybe we can pass the Embrace version and do the call outside of the component.

**Review by lucaslabari — APPROVED** [review]

**Review by SolArabehety — APPROVED** [review]

File: src/main/kotlin/io/embrace/android/intellij/plugin/ui/components/EmbBlockCode.kt [hunk 4]
```diff
@@ -1,10 +1,14 @@
...
+import io.embrace.android.intellij.plugin.dataproviders.EmbraceIntegrationDataProvider
+import io.embrace.android.intellij.plugin.network.HttpService
 import java.awt.Color
 import java.awt.Font
 import javax.swing.JLabel
 
-internal class EmbBlockCode(block: CODE_BLOCK) : JLabel() {
+internal class EmbBlockCode(block: CODE_BLOCK, httpService: HttpService) : JLabel() {
+
+    private val embraceIntegrationDataProvider = EmbraceIntegrationDataProvider(httpService)
```
Comment (lucaslabari) [inline src/main/kotlin/io/embrace/android/intellij/plugin/ui/components/EmbBlockCode.kt:11]: I think we can inject this too

**Pull request merged**